compiler/rustc_hir_typeck/src/fn_ctxt/_impl.rs

```diff
@@ -853,6 +853,13 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
         let item_name = item_segment.ident;
         let result = self
             .resolve_fully_qualified_call(span, item_name, ty.normalized, qself.span, hir_id)
+            .and_then(|r| {
+                // lint bare trait if the method is found in the trait
+                if span.edition().rust_2021() && let Some(mut diag) = self.tcx.sess.diagnostic().steal_diagnostic(qself.span, StashKey::TraitMissingMethod) {
+                    diag.emit();
+                }
+                Ok(r)
+            })
```

Comment:
Isn't this redundant with the code below?

Reply:
I don't think so. This code emits steal and emit the diag when it is able to find the target method for `dyn MyTrait`. The code below is executed only when it is not able to find such a method.